### PR TITLE
add copy assets to serve builder

### PR DESCRIPTION
**Pull request by ZachJW34**
This PR adds the copying of assets the same as how the browser builder handles assets. This behavior is in line with how the [Angular CLI](https://github.com/angular/angular-cli/blob/9d333498c6cf762b840f45b2e744c4dcf742b186/packages/angular_devkit/build_angular/src/dev-server/index.ts#L120) handles static assets when serving an application. The [Vue CLI](https://github.com/vuejs/vue-cli/blob/36f961e43dc76705878659247b563e2af83138ce/packages/%40vue/cli-service/lib/commands/serve.js#L179) handles assets via the `contentBase` property, but we need to support all assets copied from the `assets` configuration option which might contain assets from more than one directory.

### CODE REVIEW

File: libs/vue-plugin/src/builders/dev-server/builder.ts [hunk 1]
```diff
@@ -72,13 +98,14 @@ export function runBuilder(
           });
         }
       },
+      outputDir: getSystemPath(
```
Comment (BuckyMaler) [inline libs/vue-plugin/src/builders/dev-server/builder.ts:101]: Why is this required?

Comment (ZachJW34) [reply libs/vue-plugin/src/builders/dev-server/builder.ts:101]: `modifyCopyAssets` copies the assets to `options.outputPath` so we either have to change where they are copied or specify the outputDir. I went with the latter because it's nice having the build and serve builders mirror each other.
